src/NUnitFramework/framework/Attributes/TestCaseSourceAttribute.cs

```diff
@@ -173,63 +173,51 @@ private IEnumerable<ITestCaseData> GetTestCasesFor(IMethodInfo method)
 
                         if (parms == null)
                         {
-                            // 3. An array was passed, it may be an object[]
-                            //    or possibly some other kind of array, which
-                            //    TestCaseSource can accept.
```

Comment:
Is this invariant no longer true? If it is, please leave in the comment. If something else is true, add another commment.

This code has been very confusing to a lot of devs, which is why I added the comments. I'd like us to make it clear __exactly__ what we support here and what we don't.